cesm/nuopc_cap_share/init_pio_mod.F90

```diff
@@ -14,56 +15,17 @@ module shr_pio_mod
 #include <mpif.h>
 #endif
   private
-  public :: shr_pio_init
-  public :: shr_pio_component_init
-  public :: shr_pio_getiosys
-  public :: shr_pio_getiotype
-  public :: shr_pio_getioroot
-  public :: shr_pio_finalize
-  public :: shr_pio_getioformat
-  public :: shr_pio_getrearranger
-  public :: shr_pio_log_comp_settings
-
-  interface shr_pio_getiotype
-     module procedure shr_pio_getiotype_fromid, shr_pio_getiotype_fromname
-  end interface
-  interface shr_pio_getioformat
-     module procedure shr_pio_getioformat_fromid, shr_pio_getioformat_fromname
-  end interface
-  interface shr_pio_getiosys
-     module procedure shr_pio_getiosys_fromid, shr_pio_getiosys_fromname
-  end interface
-  interface shr_pio_getioroot
-     module procedure shr_pio_getioroot_fromid, shr_pio_getioroot_fromname
-  end interface
-  interface shr_pio_getindex
-     module procedure shr_pio_getindex_fromid, shr_pio_getindex_fromname
-  end interface
-  interface shr_pio_getrearranger
-     module procedure shr_pio_getrearranger_fromid, shr_pio_getrearranger_fromname
-  end interface
-
-  type pio_comp_t
-     integer :: compid
-     integer :: pio_root
-     integer :: pio_stride
-     integer :: pio_numiotasks
-     integer :: pio_iotype
-     integer :: pio_rearranger
-     integer :: pio_netcdf_ioformat
-     logical :: pio_async_interface
-  end type pio_comp_t
-
-  character(len=16), allocatable :: io_compname(:)
-  type(pio_comp_t), allocatable :: pio_comp_settings(:)
-  type (iosystem_desc_t), allocatable, target :: iosystems(:)
+  public :: init_pio_init
+  public :: init_pio_component_init
+  public :: init_pio_finalize
+  public :: init_pio_log_comp_settings
+
   integer :: io_comm
-  logical :: pio_async_interface
-  integer, allocatable :: io_compid(:)
   integer :: pio_debug_level=0, pio_blocksize=0
   integer(kind=pio_offset_kind) :: pio_buffer_size_limit=-1
 
   type(pio_rearr_opt_t) :: pio_rearr_opts
+  logical, allocatable :: pio_async_interface(:)
```

Comment:
In order to keep `pio_comp_t` the same for all drivers, I removed `pio_async_interface` from that derived type, and made that its own variable specific to the nuopc version (the mct version uses a scalar, the nuopc version has per-component values).

This change will intersect with https://github.com/ESCOMP/CMEPS/pull/305; I will make a separate comment about what needs to be done to reconcile these PRs.

Alternatively, I could put `pio_async_interface` in the shared derived type and just leave this unset by the non-nuopc versions. (I made it local because it looked like it wouldn't be needed after initialization, but if it will be needed after initialization, then this would be a better approach.)